drivers/dma/dma_mcux_lpc.c

```diff
@@ -188,10 +190,43 @@ static int dma_mcux_lpc_configure(const struct device *dev, uint32_t channel,
 
 	LOG_DBG("channel is %d", p_handle->channel);
 
-	if (config->source_chaining_en && config->dest_chaining_en) {
-		LOG_DBG("link dma out 0 to channel %d", config->linked_channel);
-		/* Link DMA_OTRIG 0 to channel */
-		INPUTMUX_AttachSignal(INPUTMUX, 0, config->linked_channel);
+	if (config->source_chaining_en || config->dest_chaining_en) {
+		if (!k_sem_take(&is_trig_mux_in_use, K_MSEC(200))) {
+			LOG_DBG("link dma out 0 to channel %d", config->linked_channel);
+			/* At the moment we only have support
+			 * for OTrig and are planning to add
+			 * support for X amount of OTrigs that
+			 * each SOC supports that use DMA_MCUX_LPC
+			 */
+
+			/* Since INPUTMUX handles the signal for
+			 * channel chaining, we should request that
+			 * the trigger is handled via INPUTMUX hardware
+			 * trigger by default
+			 */
+			DEV_BASE(dev)->CHANNEL[config->linked_channel].CFG |=
+				DMA_CHANNEL_CFG_HWTRIGEN_MASK;
+			DMA_EnableChannel(DEV_BASE(dev), config->linked_channel);
+#if CONFIG_SOC_LPC55S69_CPU0
+			/* Link DMA_OTRIG 0 to channel */
+			INPUTMUX_AttachSignal(INPUTMUX, 0,
+				kINPUTMUX_Dma0Hash0TxTrigoutToTriginChannels
+				+ channel);
+			INPUTMUX_AttachSignal(INPUTMUX, config->linked_channel,
+					kINPUTMUX_Otrig0ToDma0);
+#else
+			k_sem_give(&is_trig_mux_in_use);
+			LOG_ERR("\nAt this moment, this driver does not support"
+			"\n channel chaining for this SOC -> %s\n", CONFIG_SOC);
```

Comment:
remove newlines, this will mess up log files, there should be one log per line

Reply:
Which log files mess up with the addition of a newline character?

Reply:
any time the output is logged

Reply:
While I disagree that this breaks the log message, I have taken out the new line symbols for all instances of the log messages.

Reply:
I cannot find another example in the whole zephyr tree of anyone putting a newline in the middle of a log message